lms/djangoapps/discussion_api/api.py

```diff
@@ -208,3 +214,46 @@ def get_comment_list(request, thread_id, endorsed, page, page_size):
 
     results = [CommentSerializer(response, context=context).data for response in responses]
     return get_paginated_data(request, results, page, num_pages)
+
+
+def create_thread(request, thread_data):
+    """
+    Create a thread.
+
+    Parameters:
+
+        request: The django request object used for build_absolute_uri and
+          determining the requesting user.
+
+        thread_data: The data for the created thread.
+
+    Returns:
+
+        The created thread; see discussion_api.views.ThreadViewSet for more
+        detail.
+    """
+    course_id = thread_data.get("course_id")
+    if not course_id:
+        raise ValidationError({"course_id": ["This field is required."]})
+    try:
+        course_key = CourseLocator.from_string(course_id)
+        course = _get_course_or_404(course_key, request.user)
+    except (Http404, InvalidKeyError):
+        raise ValidationError({"course_id": ["Invalid value."]})
```

Comment:
I should also note that this message is copied from what DRF returns if field validation fails.